src/schematools/importer/base.py

```diff
@@ -437,6 +437,10 @@ def table_factory(
 
                     # We need a 'through' table for the n-m relation
                     sub_columns = [
+                        Column(
+                            "id",
+                            String,
```

Comment:
Not a fan of `String` without a `length`. String maps to `VARCHAR`. In most databases that requires a length to be specified. If we don't care about the length we should use `Text`, which maps to `TEXT`: the canonical PostgreSQL type for textual data without a length.

* https://docs.sqlalchemy.org/en/13/core/type_basics.html?highlight=string#sqlalchemy.types.String
* https://docs.sqlalchemy.org/en/13/core/type_basics.html?highlight=string#sqlalchemy.types.Text

BTW I do realize all existing code uses `String` instead of `Text`. Just wanted to have it pointed out once.

Reply:
I think in this case I'll use a fixed length varchar. This column also needs an index (reminds me, I have to check on that). Using `TEXT` does not feel right here.

Reply:
Hm. On second though, PostgreSQL docs say:

    VARCHAR (without the length specifier) and TEXT are equivalent.

I'll change it to `Text`, that make the intent most clear. As far as I can see, there is no problem with adding an index to `Text` type columns.